websites/C/Craiyon/presence.ts

```diff
@@ -8,113 +8,135 @@ type State = "start" | "generation" | "results";
-					const imageURLs = [...container.children].map(child => {
-						return (child.firstElementChild as HTMLImageElement).src;
-					});
-					for (const [i, imageURL] of imageURLs.entries()) {
+
+					for (const [i, image] of imageContainers.entries()) {
 						const presenceDataCopy = Object.assign({}, presenceData);
-						presenceDataCopy.largeImageKey = await uploadImage(imageURL);
+						presenceDataCopy.largeImageKey = image.src;
 						slideshow.addSlide(`image${i}`, presenceDataCopy, 5000);
 					}
-				} else {
-					if (slideshow.getSlides().length > 0) slideshow.deleteAllSlides();
-					presenceData.details = "Viewing a generated image";
-					presenceData.state = `"${oldPrompt}"`;
-					presenceData.largeImageKey = await uploadImage(
-						container.querySelector("img").src
-					);
+					useSlideshow = true;
 				}
 			}
 			break;
 		}
-		case "/privacy": {
+		case "image": {
+			presenceData.details = "Viewing an image";
+			presenceData.state = (
+				document
+					.evaluate("//p[text()='Prompt']", document.body)
+					.iterateNext() as HTMLParagraphElement
+			).nextElementSibling;
+			presenceData.largeImageKey =
+				document.querySelector<HTMLImageElement>("main img").src;
+			presenceData.buttons = [
+				{
+					label: "View Image Details",
+					url: href,
+				},
+			];
+			break;
+		}
+		case "search": {
+			if (pathList.length > 1) {
+				presenceData.details = "Viewing search results";
+				presenceData.state = document.querySelector("input").value;
+				const now = Date.now();
+				if (now - searchResultCacheTimestamp > 5000) {
+					searchResultCacheTimestamp = now;
+
+					for (const [i, image] of document
+						.querySelectorAll<HTMLImageElement>("main a img")
+						.entries()) {
+						const presenceDataCopy = Object.assign({}, presenceData);
+						presenceDataCopy.largeImageKey = image.src;
+						slideshow.addSlide(`image${i}`, presenceDataCopy, 5000);
+					}
+				}
+				useSlideshow = true;
+			} else presenceData.details = "Searching for images";
+			break;
+		}
+		case "blog": {
+			if (pathList.length > 1) {
+				presenceData.details = "Reading a blog post";
+				presenceData.state = document.querySelector("h1");
+				presenceData.largeImageKey =
+					document.querySelector<HTMLImageElement>("main p + img").src;
+				presenceData.buttons = [
+					{
+						label: "View Blog Post",
+						url: href,
+					},
+				];
+			} else presenceData.details = "Browsing blog posts";
+			break;
+		}
+		case "privacy": {
 			presenceData.details = "Reading privacy policy";
 			break;
 		}
-		case "/terms": {
+		case "terms": {
 			presenceData.details = "Reading terms and conditions";
 			break;
 		}
 	}
 	if (presenceData.details) {
-		if (slideshow.getSlides().length > 0) presence.setActivity(slideshow);
-		else presence.setActivity(presenceData);
+		if (useSlideshow) {
+			if (!slideshow.currentSlide.details)
+				slideshow.currentSlide = slideshow.getSlides()[0].data;
```

Comment:
https://discord.com/channels/493130730549805057/1188944048728309770/1188944048728309770
Will convert to draft for now